pipeline/test_bluesky_extract.py

```diff
@@ -0,0 +1,229 @@
+"""Test script for BlueSky extract python file."""
+
+
+from unittest.mock import patch, MagicMock
+from freezegun import freeze_time
+import os
+import io
+import logging
+import pytest
+import csv
+from unittest.mock import MagicMock, patch
+from atproto import models
+from atproto_firehose import FirehoseSubscribeReposClient
+from bluesky_extract import (format_text, extract_text_from_bytes,
+                             get_firehose_data, connect_and_write,
+                             OUTPUT_FOLDER, HEADER)
+
+
+@pytest.fixture
+def sample_bytes():
+    return b'''{"text": "Hello World", "$type": "app.bsky.feed.post", "langs": ["en"], "createdAt": "2024-12-03T11:17:35.355Z"}'''
+
+
+def test_correct_formatting():
+    """Test the function removes all extra new lines and whitespaces successfully."""
+    test_string = "H ello    \n \n \n Good bye!"
+    assert format_text(test_string) == "H ello Good bye!"
+
+
+@patch('bluesky_extract.json')
+@patch('bluesky_extract.logging')
+def test_successful_extract_text(mock_logging, mock_json, sample_bytes):
+    """Parsed = dict, json_data=str, text=str"""
+    """Test that bytes are successfully parsed into dict then strings."""
+
+    raw_byte_input = sample_bytes
+    mock_json.dumps.return_value = '{"text": "Hello World", "$type": "app.bsky.feed.post", "langs": ["en"], "createdAt": "2024-12-03T11:17:35.355Z"}'
+    mock_json.loads.return_value = {"text": "Hello World", "$type": "app.bsky.feed.post", "langs": [
+        "en"], "createdAt": "2024-12-03T11: 17: 35.355Z"}
+    result = extract_text_from_bytes(raw_byte_input)
+    assert result == "Hello World"
+    assert isinstance(result, str)
+
+
+@patch('bluesky_extract.json')
+@patch('bluesky_extract.logging')
```

Comment:
instead of having to repeat these 2 lines for multiple tests, try a fixture that returns the mocked files:
```
@pytest.fixture
def mock_logging_json():
    with patch('bluesky_extract.logging') as mock_logging, patch('bluesky_extract.json') as mock_json:
        yield mock_logging, mock_json
```

Reply:
I  will implement this on the next one!